primefaces/src/main/resources/META-INF/resources/primefaces/core/core.js

```diff
@@ -378,6 +378,62 @@
             return this;
         },
 
+        /**
+         * Applies the inline AJAX status (ui-state-loading) to the given widget / button.
+         * @param {PrimeFaces.widget.BaseWidget} [widget] the widget.
+         * @param {JQuery} [button] The button DOM element.
+         * @param {(widget: PrimeFaces.widget.BaseWidget, settings: JQuery.AjaxSettings) => boolean} [isXhrSource] Callback that checks if the widget is the source of the current AJAX request.
+         */
+        bindButtonInlineAjaxStatus: function(widget, button, isXhrSource) {
+            if (!isXhrSource) {
+                isXhrSource = function(widget, settings) {
+                    return PrimeFaces.ajax.Utils.isXhrSource(widget, settings);
+                };
+            }
+
+            widget.ajaxCount = 0;
+            $(document).on('pfAjaxSend.' + widget.id, function(e, xhr, settings) {
+                if (isXhrSource.call(this, widget, settings)) {
+                    widget.ajaxCount++;
+                    if (widget.ajaxCount > 1) {
+                        return;
+                    }
+
+                    button.addClass('ui-state-loading');
+
+                    if (widget.hasOwnProperty('disable')
```

Comment:
What does / should this line do? It is not passed for the "Ajax Disabling" buttons in the showcase.
```
console.log(widget.hasOwnProperty('disable'));
console.log(widget.cfg.disableOnAjax);
```
Results in
```
false
undefined
```

Reply:
Is it plain JS? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty ?

Reply:
How can I check this with file upload? I think we might be able to use `typeof widget.disable === 'function'`

Reply:
SHouldn't that first one be `widget.cfg.hasOwnProperty` not `widget.hasOwnProperty` ?